frontend/src/metabase/dashboard/actions/cards.js

```diff
@@ -71,6 +76,27 @@ export const addCardToDashboard =
     dispatch(loadMetadataForDashboard([dashcard]));
   };
 
+export const removeCardFromDashboard = createThunkAction(
+  REMOVE_CARD_FROM_DASH,
+  ({ dashcardId, cardId }) =>
+    (dispatch, _getState) => {
+      dispatch(cancelFetchCardData(cardId, dashcardId));
+      return { dashcardId, cardId };
+    },
+);
+
+export const undoRemoveCardFromDashboard = createThunkAction(
+  UNDO_REMOVE_CARD_FROM_DASH,
+  ({ dashcardId }) =>
+    (dispatch, getState) => {
+      const dashcard = getDashCardById(getState(), dashcardId);
+      const card = dashcard.card;
+
+      dispatch(fetchCardData(card, dashcard));
```

Comment:
Does it mean that we will always re-fetch card data on restoring the card? Was it like that before?

Reply:
Good point, I didn't think about that implication.
I added a simple check to see if we already have the data in redux before fetching it again, let me know if there are smarter ways to do it.